File: 0600-0699/0617/0617.java
class TreeNode {
	int val;
	TreeNode left;
	TreeNode right;
	TreeNode() {}
	TreeNode(int val) { this.val = val; }
	TreeNode(int val, TreeNode left, TreeNode right) {
		this.val = val;
		this.left = left;
		this.right = right;
	}
}

class Solution {
	public TreeNode mergeTrees(TreeNode root1, TreeNode root2) {
		if (root1 == null && root2 == null) {
			return null;
		}
		return new TreeNode(
			(root1 != null ? root1.val : 0) + (root2 != null ? root2.val : 0),
			// Merge leaf nodes recursively
			mergeTrees(root1 != null ? root1.left : null, root2 != null ? root2.left : null),
			mergeTrees(root1 != null ? root1.right : null, root2 != null ? root2.right : null)
		);
	}
}

File: 0600-0699/0617/0617.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
	def mergeTrees(self, root1, root2):
		"""
		:type root1: TreeNode
		:type root2: TreeNode
		:rtype: TreeNode
		"""
		if not root1:
			return root2
		if not root2:
			return root1
		return TreeNode(
			(root1.val if root1 else 0) + (root2.val if root2 else 0),
			# Merge leaf nodes recursively
			self.mergeTrees(root1.left if root1 else None, root2.left if root2 else None),
			self.mergeTrees(root1.right if root1 else None, root2.right if root2 else None)
		)